src/core/Form/FileInput/FileInput.tsx

```diff
@@ -0,0 +1,603 @@
+  /** CSS class for custom styles */
+  className?: string;
+  /** Label for the input */
+  labelText: ReactNode;
+  /** Hides or shows the label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: LabelMode;
+  /** Text to show in the file selection button. Example: "Choose file" */
+  inputButtonText: ReactNode;
+  /** Text to show inside the file drag area */
+  dragAreaText: ReactNode;
+  /** Text to show in the button which removes a file from the input */
+  removeFileText: ReactNode;
+  /** Allow preview of files */
+  filePreview?: boolean;
+  /** Text for assistive technology to be prepended to added file names. E.g. "Added file: " */
+  addedFileAriaText: string;
+  /** Callback fired on input click */
+  onClick?: () => void;
+  /** Callback fired on input change */
+  onChange?: (value: FileList) => void;
+  /** Callback fired on input blur */
+  onBlur?: () => void;
+  /** Controlled value of the input */
+  value?: ControlledFileItem[];
+  /** Hint text to be shown below the component's label. Should be used to give file format instructions */
+  hintText?: string;
+  /**
+   * `'default'` | `'error'`
+   *
+   * Status of the component. Error state creates a red border around the input.
+   * Always use a descriptive `statusText` with an error status.
+   * @default default
+   */
+  status?: InputStatus;
+  /** Toggles small screen styling */
+  smallScreen?: boolean;
+  /**
+   * HTML id attribute.
+   * If no id is specified, one will be generated automatically
+   */
+  id?: string;
+  /** HTML name attribute for the input */
+  name?: string;
+  /**
+   * aria-describedby for the HTML input element,
+   */
+  'aria-describedby'?: string;
+  /** Sets component's width to 100% of its parent */
+  fullWidth?: boolean;
+  /** Text to mark the field optional. Will be wrapped in parentheses and shown after `labelText` */
+  optionalText?: string;
+  /** Inline CSS styles for the component's outer layer */
+  style?: React.CSSProperties;
+  /** Tooltip component for the input's label */
+  tooltipComponent?: ReactElement;
+}
+
+type MultiFileProps =
+  | {
+      multiFile?: false;
+      multiFileListHeadingText?: string;
+    }
+  | {
+      /** Allow multiple files in the input
+       * @default false
+       */
+      multiFile?: true;
+      /** Text to show above the file listing. Required with `multiFile` */
+      multiFileListHeadingText: string;
+    };
+
+export type FileInputProps = BaseFileInputProps & MultiFileProps;
+
+export interface FileItemRefs {
+  id: string;
+  fileNameRef: React.RefObject<HTMLAnchorElement | HTMLDivElement>;
+  fileSizeElementId: string;
+}
+
+export interface ControlledFileItem {
+  file: File;
+  status?: 'default' | 'error' | 'loading';
+  errorText?: string;
+  ariaLoadingText?: string;
+  buttonText?: string;
+  buttonIcon?: ReactElement;
+  buttonOnClick?: (file: File) => void;
```

Comment:
These could benefit from documentation since these are exported and used outside the library as well.

Reply:
Could be an almost direct copypaste from the styleguidist controlled state example.

Reply:
Comments added 👍

Reply:
I think these need to be the jsdoc format in order to enable the tooltip inside vscode. So the comment format should be `/**  The actual file object */` instead of `// The actual file object`.

Reply:
Ah yes, fixed this!